### ISSUE TITLE
Create a JSON endpoint(s) to get Brigade-specific stats

### ISSUE THREAD

**Issue by zachmccormick** [code-for-nashville/brigade-pulse#59]
The Brigade-stats page should display the following information:
- Number of meetups per day/week/month (line graph)
- Size of meetup group per day/week/month (line graph)
- Number of commits per day/week/month to all organization repos (line graph)
- Number of issues opened per day/week/month for all organization repos (line graph)
- Number of issues closed per day/week/month for all organization repos (line graph)
- Current language distribution for all organization repos by bytes/LoC (bar graph)
- Current LoC in HEAD of master branch by contributor for all organization repos (bar graph)
- List of repos/projects for the brigade (clickable to per-project stats page)

The JSON endpoint(s) should output the appropriate data in order to make the rendering of this page possible.

**Comment by neolytics** [comment]
Hey @zachmccormick I'm checking on the endpoint stuff here.  Can you elaborate on the desired format of this or point me to where the desired implementation should be.  It's actually nice to have someone else doing python work here so I can make contributions like a normal human.

**Comment by zachmccormick** [comment]
Hehe no problem!  Honestly this is probably blocked by frontend work at the moment - just need to rework the brigade details page a little bit to display this information a little better. [Here's the details page](http://nashville-brigade-pulse.herokuapp.com/#/brigades/Code-for-Nashville) - I think we should shrink the text on these things and have paginated tables for events + projects.  On the top of the page maybe a fixed size table for Meetup Group stats like Organizer Name, Topics, Members, Rating.

**Comment by zachmccormick** [comment]
Once we actually have some real time-series data we can generate graphs and put them on this page as necessary too

I think this endpoint will be ultimately dictated by what we display on this page.

**Comment by zachmccormick** [comment]
Maybe it will look something like this

![sketch](https://cldup.com/oCIgOxN3cV-3000x3000.png)

**Comment by mshenfield** [comment]
This looks pretty good. I can make an example HTML page using static data. I'll use Highcharts and if we don't like it or need something more custom we can switch it out.

**Comment by mshenfield** [comment]
Hey, created a rough draft of the updated details page at branch update-brigade-details-with-json-endpoints 51e58a41941a22318d3994c5d3178d5d356254f1. I'm using static JSON objects in the brigade.controller.js instead of loading data from an HTTP endpoint, we can switch that out and add the client-side processing once we get the backend API built out.

This branch also includes a few independent formatting tweaks which I'm going to cherrypick back into master when I get the chance today or tomorrow.

**Comment by zachmccormick** [comment]
Woo hoo!!!

-Zach

--
Zach McCormick
317-443-2799

**Comment by zachmccormick** [comment]
@mshenfield how's the change to that design going?  I know @beck410 was interested in helping with some front-end stuff as well!  I may tomorrow or Sunday make a rough draft of what I think the top level page should look like as well (this being the second-level page).

My gut instinct for time's sake is to have a top level page, where you can view/sort brigades using DataTables, and this second-level page, with stats and graphs.  I think that would be a good 'alpha' version for Jon to show off in Oakland!

**Comment by zachmccormick** [comment]
@mshenfield any update on that design?  I might check it out later today + make an endpoint/push it live if it works!

**Comment by zachmccormick** [comment]
I made the endpoint for this but need to implement the Projects + Events lists.  The GitHub repos graph data is messed up in the database unfortunately, so if anyone has a suggestion to replace that graph for now, I'm game!
